text/0096-metric-instrument-terminology.md

```diff
@@ -0,0 +1,153 @@
+identified four kinds of "refinement" and mapped out the space of
+_possible_ instruments, while not proposing which would actually be
+included in the standard.
+
+[OTEP 93](https://github.com/open-telemetry/oteps/pull/93) followed
+with a list of six standard instruments, the most necessary and useful
+combination of instrument refinements, plus one special case used to
+record timing measurements.
+
+This proposal finalizes the names used to describe the seven
+instruments above, seeking to address core confusion related to
+"Measure":
+
+1. OTEP 88 stipulates that the terms currently in use to name
+synchronous and asynchronous instruments become abstract, but also
+using "Measure-like" and "Observer-like" to discuss instruments with
+refinements.  This proposal states that we shall prefer the
+adjectives, commonly abbreviated "Sync" and "Async", when describing
+instruments.
+2. Prior to OTEP 88, but even with OTEPs 88 and 93 included, there is
+inconsistency in the naming of instruments.  Note that "Counter" and
+"Observer" end in "-er", a noun suffix used in the sense of "[person
+occupationally connected
+with](https://www.merriam-webster.com/dictionary/-er)", while the term
+"Measure" does not fit this pattern.  This proposal proposes to
+replace the abstract term "Measure" by "Recorder", since the
+associated method name (verb) is specified as `Record()`.
+3. The OTEP 93 asynchronous instruments ("LastValueObserver",
+"DeltaObserver", and "CumulativeObserver") have the pattern
+"-Observer", while the OTEP 93 synchronous instruments
+("Counter", "UpDownCounter", "Distribution", "Timing") do not.  This
+proposal keeps "Counter" and "UpDownCounter" for Sum-only synchronous instruments, and does the same
+with "Recorder", yielding "Recorder" and "TimingRecorder".
+4. Confusion over the loss of "Gauge" is addressed by replacing
+"LastValueObserver" with "GaugeObserver".
+
+This proposal also repeats the current specification of the default
+Aggregator for each kind of instrument.
+
+## Explanation
+
+The following table summarizes the four synchronous instruments and
+three asynchronous instruments that will be standardized as a result
+of this set of proposals.
+
+| Existing name | OTEP 93 name       | **Final name**         | Sync or Async | Function | Default aggregation | Rate support |
+| ------------- | ------------------ | ---------------------- | ----------- | ------------- | ---------- | ---- |
+| Counter       | Counter            | **Counter**            | Sync  | Add() | Sum | Yes | 
+|               | UpDownCounter      | **UpDownCounter**      | Sync  | Add() | Sum | Yes |
+| Measure       | Distribution       | **Recorder**           | Sync  | Record() | MinMaxSumCount | No |
+|               | Timing             | **TimingRecorder**     | Sync  | Record() | MinMaxSumCount  | No |
+| Observer      | LastValueObserver  | **GaugeObserver**      | Async | Observe() | MinMaxSumCount | No |
```

Comment:
I really want this to be `Sum` because of memory-usage metric. Look at the Go example for process metrics https://github.com/open-telemetry/opentelemetry-specification/issues/549#issue-595864966 I don't want to see "process/sys_heap" as a minmaxsumcount by default.

Another argument is default cost (money), if I am using a system like Stackdriver exporting "minmaxsumcount" will cost 4 times more than just sum. So I think it is important that default aggregations are less "expensive".

PS: I know I work for a company that makes money if the user sends more data :)

Reply:
Firstly, I think minmaxsumcount is perfectly sensible for a heap. If you aggregate this across a cluster, you'd like to know what the min/max values are.  If you don't aggregate the count as well, the metric system will requires domain-level knowledge to construct an average using some other knowledge (i.e.,, that there's a corresponding count available, implicitly).

Moreover, I've documented (here) how there's an obvious optimization that results in sending just one value for the min/max/sum/count when there is just one value, which reduces the data-size problem for the case you care about.  Each machine will report one min/max/sum/count as a single value in the ordinary case, but when we aggregate those values, we'll get the proper min/max/sum and count out--no domain-knowledge required.

Lastly, the metric instrument you're describing is exactly the hypothetical `UpDownCumulativeObserver` detailed in OTEP 88.  If you want a Sum-only aggregation, then you should use a Sum-only instrument. That instrument does exactly what you want, should we standardize it?